dspy/primitives/module.py

```diff
@@ -26,10 +26,8 @@ def add_parameter(param_name, param_value):
                 add_parameter(name, value)
 
             elif isinstance(value, BaseModule):
-                # When a sub-module is pre-compiled, keep it frozen.
-                if not getattr(value, '_compiled', False):
```

Comment:
Ok but now we'll need to modify bootstrap.py at least so it replicates this behavior.

I guess that means: freeze the module that's like that...

Reply:
Sure, I will add this.

Reply:
I've resolved this here: https://github.com/stanfordnlp/dspy/commit/8f4a0e5ea42c8e22fc8b3180c7924becee975ec4

Reply:
By default named_parameters should return all parameters, so I added an only_uncompiled flag and made sure all named_parameter calls in bootstrapping / finetuning set this flag to true.